holmes/plugins/toolsets/coding_agent/coding_agent_toolset.py

```
@@ -0,0 +1,534 @@
+            capture_output=True,
+            text=True,
+            timeout=120,
+        )
+    except subprocess.CalledProcessError as e:
+        shutil.rmtree(workdir, ignore_errors=True)
+        raise RuntimeError(
+            f"Failed to clone {repo_url} (branch: {base_branch}): {e.stderr}"
+        ) from e
+    except subprocess.TimeoutExpired:
+        shutil.rmtree(workdir, ignore_errors=True)
+        raise RuntimeError(f"Clone of {repo_url} timed out after 120 seconds")
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)